Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla44074.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Diagnostics;
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+
+#if UITEST
+using Xamarin.UITest;
+using NUnit.Framework;
+#endif
+
+// Apply the default category of "Issues" to all of the tests in this assembly
+// We use this as a catch-all for tests which haven't been individually categorized
+#if UITEST
+[assembly: NUnit.Framework.Category("Issues")]
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Bugzilla, 44074, "OnDetachingFrom for Behavior is never called, therefore memory can leak due to event handlers not being unbound", PlatformAffected.All)]
+	public class Bugzilla44074 : TestNavigationPage // or TestMasterDetailPage, etc ...
+	{
+		protected override void Init()
+		{
+			var cp = new ContentPage();
+			var sl = new StackLayout { Padding = 20 };
+
+			var label = new Label
+			{
+				Text = "This is the first page with no behaviors",
+				HorizontalTextAlignment = TextAlignment.Center,
+				VerticalTextAlignment = TextAlignment.Center
+			};
+			sl.Children.Add(label);
+
+			var button = new Button
+			{
+				Text = "Open Behavior View",
+				Command = new Command(() => { Navigation.PushAsync(new TestSecondPage()); })
+			};
+			sl.Children.Add(button);
+
+			cp.Content = sl;
+			PushAsync(cp);
```

Comment:
expecting the behaviors to be detached t this point might be a valid assumption, except that you have to take into account the unpredictable behavior of the Garbage Collector, or collectors.

e.g., in the case of the Mono GC, the TestSecondPage is probably not collect yet at this point.